# ReadMe
1. Data Preparation
- Import 'master_data.csv' which has been prepared in 'COMP4135_Data_Preparation.ipynb'

2. Multiple Regression
- Use four independent variables at once to predict a dependent variable <br>

3. Independent Variable
- Average Watching Count 
- Video Completion Percentage 
- Enrolled Course Count 
- Average Enroll Time

4. Dependent Variable
- Grade

## Multiple Regression 

In [1]:
import numpy as np 
import pandas as pd
import matplotlib.pyplot as plt

In [2]:
#1 Import dataset
master_data = pd.read_csv('master_data.csv')

In [3]:
#2 Make a list of the independent values (X) and dependent values in a variable (Y).

# X = four features
# y = grade

X = master_data.iloc[:, 2:6].values
y = master_data.iloc[:, 6].values

In [4]:
#3 Splitting the dataset into the Training and Test dataset

from sklearn.model_selection import train_test_split
X_train, X_test, y_train, y_test = train_test_split(X, y, test_size=0.2, random_state=0)

In [5]:
#4 Create a multiple Linear Regression object

from sklearn.linear_model import LinearRegression
multiregr = LinearRegression()

In [6]:
#5 Fit the linear regression model to the training set
multiregr.fit(X_train, y_train)

LinearRegression()

In [7]:
#6 Predict the Test set results

y_pred= multiregr.predict(X_test)
y_pred

array([49.33547507, 49.4865026 , 49.50921054, ..., 49.30723742,
       49.59492726, 48.88469507])

In [8]:
y_test

array([89, 20, 37, ..., 86, 88, 95], dtype=int64)

In [9]:
# correlation of 4 features
print("Correlation of four features = {}".format(multiregr.coef_))

Correlation of four features = [-0.04752545 -0.39964997 -0.02335711 -0.01576584]


<b>Interpretation of Correlation</b>
<li> One more time of watching video: 0.04752545 DECREASE in grade
<li> One more percent of completing video: 0.39964997 DECREASE in grade
<li> One more registered course: 0.02335711 DECREASE in grade
<li> One hour late for enrolling courses: 0.01576584 DECREASE in grade

## Demonstration

In [10]:
# predict student's grade whose
    # average watching count = 20
    # average completion percentage = 0.5 (50%)
    # registered courses = 2
    # average enroll time = 23 (11pm)

predictedGrade = multiregr.predict([[7.44, 0.946, 5, 11]])
predictedGrade

array([49.31079523])